core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.policy;
+
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.client.block.policy.options.GetWorkerOptions;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.wire.WorkerNetAddress;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.TreeMap;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Random load block base on capacity of worker,
+ * used for cluster of workers with different capacity.
+ * Then larger capacity worker will receive more load request.
+ *
+ */
+@ThreadSafe
+public class CapacityBaseRandomPolicy implements BlockLocationPolicy {
+
+  /**
+   * Constructs a new {@link CapacityBaseRandomPolicy}.
+   *
+   * @param conf Alluxio configuration
+   */
+  public CapacityBaseRandomPolicy(AlluxioConfiguration conf) {
+  }
+
+  @Nullable
+  @Override
+  public WorkerNetAddress getWorker(GetWorkerOptions options) {
+    Iterable<BlockWorkerInfo> blockWorkerInfos = options.getBlockWorkerInfos();
+    //key is the floor of capacity range the worker cover
+    TreeMap<Long, BlockWorkerInfo> capacityFloorMap = new TreeMap<>();
+    AtomicLong totalCapacity = new AtomicLong(0L);
+    blockWorkerInfos.forEach(workerInfo -> {
+      if (workerInfo.getCapacityBytes() > 0) {
+        long capacityRangeFloor = totalCapacity.getAndAdd(workerInfo.getCapacityBytes());
+        capacityFloorMap.put(capacityRangeFloor, workerInfo);
+      }
+    });
+    if (totalCapacity.get() == 0L) {
+      return null;
+    }
+    long randomLong = randomInCapacity(totalCapacity.get());
+    return capacityFloorMap.floorEntry(randomLong).getValue().getNetAddress();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    return this == o || o instanceof CapacityBaseRandomPolicy;
+  }
+
+  @Override
+  public int hashCode() {
+    return 0;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this).toString();
+  }
```

Comment:
These implementations are a bit strange, especially the hashCode returns 0 so this is purely dummy. However I get it that you are doing the same as `MostAvailableFirstPolicy`. Also I don't find any places that rely on the hashCode of `BlockLocationPolicy` so I think it's fine.

Reply:
Right, let's not supply `hashCode` and `equals` with dummies and can just drop those to revert behavior to java default.

Also, `toStringHelper` with no fields is also as good as not providing a `toString()` implementation at all.